nilearn/surface/surface.py

```diff
@@ -1592,6 +1593,34 @@ def __getitem__(self, index):
                 "Index out of range. Use 0 for coordinates and 1 for faces."
             )
 
+    @property
+    def _area(self):
+        """Compute area of mesh.
+
+        Get the vertex coordinates for each face
+         Compute vectors for two edges of the triangle
+         Compute the cross product of the two edge vectors
+         Area of triangle = 0.5 * norm of cross product
+        """
+        total_area = 0.0
+        for face in self.faces:
+            v0, v1, v2 = (
+                self.coordinates[face[0]],
+                self.coordinates[face[1]],
+                self.coordinates[face[2]],
+            )
+
+            edge1 = v1 - v0
+            edge2 = v2 - v0
+
+            cross_prod = np.cross(edge1, edge2)
+
+            area = 0.5 * np.linalg.norm(cross_prod)
+
+            total_area += area
```

Comment:
this may be very slow
one possible work around for most common cases that involve freesurfer surfaces:
- pre-compute the area of freesurfer surfaces
- check if the mesh is one of the ones known to nilearn
- return the precomputed area if so

Reply:
Have you timed it ? Maybe it's possible to vectorize the whole computation ?